### PR TITLE
Create chain tx

### PR DESCRIPTION
**Pull request by kmenon01**
Finished working implementation of Create Chain Transaction

### CODE REVIEW

File: src/parser.c [hunk 1]
```diff
@@ -485,7 +500,7 @@ enum parse_rv parse_SubnetAuth(struct SubnetAuth_state *const state, parser_meta
     case 0:
       // Type ID
       CALL_SUBPARSER(uint32State, uint32_t);
-      PRINTF("SUBNET AUTH\n");
+      //PRINTF("SUBNET AUTH\n");
```
Comment (patrick-ogrady) [inline src/parser.c]: Should we remove instead of commenting?

File: src/parser.c [hunk 2]
```diff
@@ -520,13 +535,12 @@ enum parse_rv parse_SubnetAuth(struct SubnetAuth_state *const state, parser_meta
         if (state->sigindices_i < state->sigindices_n)
         {
           INIT_SUBPARSER(uint32State, uint32_t);
-          RET_IF_PROMPT_FLUSH;
+	  //PRINTF("Reached\n");
```
Comment (patrick-ogrady) [inline src/parser.c]: Remove?

File: src/parser.c [hunk 3]
```diff
@@ -1371,6 +1385,74 @@ enum parse_rv parse_AddSNValidatorTransaction(
...
+{
+  enum parse_rv sub_rv = PARSE_RV_INVALID;
+  switch(state->state)
+  {
+    case 0: {
+      // Number of bytes of Genesis Data
+      CALL_SUBPARSER(uint32State, uint32_t);
+      state->state++;
+      state->gen_n = state->uint32State.val;
+      PRINTF("Gen Data Count\n");
```
Comment (patrick-ogrady) [inline src/parser.c:1403]: Debug log?

File: src/parser.c [hunk 4]
```diff
@@ -1371,6 +1385,74 @@ enum parse_rv parse_AddSNValidatorTransaction(
...
+      INIT_SUBPARSER(uint8State, uint8_t);
+    } fallthrough;
+    case 1: {
+      //
```
Comment (patrick-ogrady) [inline src/parser.c]: Empty comment

File: tests/p-chain-tests.ts [hunk 5]
```diff
@@ -430,16 +431,138 @@ describe('Staking tests', async function () {
...
+      0x00, 0x00, 0x00, 0x00]);
 
     const pathPrefix = "44'/9000'/0'";
     const pathSuffixes = ["0/0", "0/1", "100/100"];
     // Need to add headers for SubnetID and Sigindices?
     const prompts = chunkPrompts([
-      {header: 'Sign', body: 'Create Subnet'},
-      {header: 'Threshold', body: '1'},
-      {header: 'Address', body: 'local1mg47uqd7stkvqrp57ds7m28txra45u2uzkta8n' },
+      {header: 'Sign', body: 'Create Chain'},
```
Comment (patrick-ogrady) [inline tests/p-chain-tests.ts:513]: This is great

File: src/parser.h [hunk 6]
```diff
@@ -389,7 +444,7 @@ typedef struct {
         //TODO: Now that weve added this, do we need the ones above?
         output_prompt_t output_prompt;
     } data;
-} prompt_entry_t;
+}prompt_entry_t;
```
Comment (patrick-ogrady) [inline src/parser.h]: Did we mean to remove this space?

File: src/parser.c (hunk 3 above)
Comment (kmenon01) [reply src/parser.c:1403]: Yes this is for the DEBUG log

**Review by patrick-ogrady — APPROVED** [review]
Nice!

**Comment by Ericson2314** [comment]
@kmenon01 Have a good look at the last commit especially. Without hashing in constant space, we will fail as badly on large genesis data as if were to store and print the entire thing out without hashing it.

**Pull request merged**